hack/update-crds.sh

```diff
@@ -40,18 +40,28 @@ REPO_ROOT=${BUILD_WORKSPACE_DIRECTORY}
 cd "${REPO_ROOT}"
 
 "$controllergen" \
-  schemapatch:manifests=./deploy/charts/cert-manager/crds \
-  output:dir=./deploy/charts/cert-manager/crds \
+  schemapatch:manifests=./deploy/charts/cert-manager-crds/templates \
+  output:dir=./deploy/charts/cert-manager-crds/templates \
   paths=./pkg/apis/...
 
 out="./deploy/manifests/00-crds.yaml"
 rm "$out" || true
 touch "$out"
-for file in $(find "./deploy/charts/cert-manager/crds" -type f | sort -V); do
-  # concatenate all files while removing blank (^$) lines
-  < "$file" sed '/^$$/d' >> "$out"
-  printf -- "---\n" >> "$out"
-done
+# template all files while removing blank (^$) lines
+bazel run //hack/bin:helm template ./deploy/charts/cert-manager-crds/ | sed '/^$$/d' >> "$out"
+printf -- "---\n" >> "$out"
 chmod 644 "$out"
 
+
 echo "Generated 00-crds.yaml"
+
+path="./deploy/manifests/crds/"
+mkdir "$path" || true
+for file in $(find "./deploy/charts/cert-manager-crds/templates/" -type f -name "*.yaml" | sort -V); do
+  # template file while removing blank (^$) lines
+  out="$path/$(basename $file)"
+  rm "$out" || true
+  touch "$out"
+  bazel run //hack/bin:helm template -s "templates/$(basename $file)"  ./deploy/charts/cert-manager-crds/ | sed '/^$$/d' >> "$out"
+
+done
```

Comment:
I'd rather avoid this step for now - it generates ~25k lines of duplicated YAML used solely for one set of tests.

I think we may be able to work around this in a few ways:

* Adding a function that allows extracting a single `v1beta1.CustomResourceDefinition` from a file, and then passing that to the `crd-schema-fuzz` package's `SchemaFuzzTestForInternalCRD` function: https://godoc.org/github.com/munnerz/crd-schema-fuzz#SchemaFuzzTestForInternalCRD.

* Adding the 'built' CRD as a 'src' dependency to the `go_test` target in this file. This has the downside of causing `go test` to fail/skip if this package is tested directly with the `go` tool.

* Invoke Bazel in the test to generate the CRD, but this has the inverse effect of then not working when run with `bazel test` 😅 

The first option seems like it is the 'cleanest' and allows the most compatibility, without duplicating all of this stuff again 😅

Reply:
Implemented the first option!